L1Trigger/Phase2L1ParticleFlow/plugins/L1CTJetFileWriter.cc

```diff
@@ -40,43 +42,82 @@ class L1CTJetFileWriter : public edm::one::EDAnalyzer<edm::one::SharedResources>
   void analyze(const edm::Event&, const edm::EventSetup&) override;
   void endJob() override;
   std::vector<ap_uint<64>> encodeJets(const std::vector<l1t::PFJet> jets);
+  std::vector<ap_uint<64>> encodeSums(const std::vector<l1t::EtSum> jets);
 
-  edm::EDGetTokenT<edm::View<l1t::PFJet>> jetsToken_;
   l1t::demo::BoardDataWriter fileWriterOutputToGT_;
+  std::vector<std::pair<edm::EDGetTokenT<edm::View<l1t::PFJet>>, edm::EDGetTokenT<edm::View<l1t::EtSum>>>> tokens_;
+  std::vector<std::pair<bool, bool>> tokensToWrite_;
+  //std::vector<edm::EDGetTokenT<edm::View<l1t::PFJet>>> jetsTokens_;
+  //std::vector<edm::EDGetTokenT<edm::View<l1t::EtSum>>> mhtTokens_;
 };
 
 L1CTJetFileWriter::L1CTJetFileWriter(const edm::ParameterSet& iConfig)
-    : nJets_(iConfig.getParameter<unsigned>("nJets")),
+    : collections_(iConfig.getUntrackedParameter<std::vector<edm::ParameterSet>>("collections",
+                                                                                 std::vector<edm::ParameterSet>())),
+      nJets_(iConfig.getParameter<unsigned>("nJets")),
       nFramesPerBX_(iConfig.getParameter<unsigned>("nFramesPerBX")),
       ctl2BoardTMUX_(iConfig.getParameter<unsigned>("TMUX")),
-      gapLengthOutput_(ctl2BoardTMUX_ * nFramesPerBX_ - 2 * nJets_),
+      gapLengthOutput_(ctl2BoardTMUX_ * nFramesPerBX_ - 2 * (nJets_ + 1) * collections_.size()),
```

Comment:
here the +1 is correct only if 2 sums are sent, but from the code below it seems the 2 sums are added only if the `mht` parameter is given.
perhaps you could have a parameter `nSums` , and change this to `(2*nJets_ + nSums_)*collections_.size()`

Reply:
This is now reimplemented such that each collection can have a different number of jets and sums, and the gap is computed consistently